pkg/util/tests/kubectl/kubectl.go

```diff
@@ -0,0 +1,304 @@
+	"strings"
+	"time"
+
+	"github.com/istio-ecosystem/sail-operator/pkg/util/tests/shell"
+	r "github.com/istio-ecosystem/sail-operator/pkg/util/tests/types"
+)
+
+const DefaultBinary = "kubectl"
+
+var (
+	ErrNotFound       = errors.New("resource was not found")
+	EmptyResourceList = r.ResourceList{
+		APIVersion: "v1",
+		Items:      []interface{}{},
+		Kind:       "List",
+		Metadata:   r.Metadata{ResourceVersion: ""},
+	}
+)
+
+// kubectl return the kubectl command
+// If the environment variable COMMAND is set, it will return the value of COMMAND
+// Otherwise, it will return the default value "kubectl" as default
+// Arguments:
+// - format: format of the command without kubeclt or oc
+// - args: arguments of the command
+func kubectl(format string, args ...interface{}) string {
+	binary := DefaultBinary
+	if cmd := os.Getenv("COMMAND"); cmd != "" {
+		binary = cmd
+	}
+
+	return binary + " " + fmt.Sprintf(format, args...)
+}
+
+// ApplyString applies the given yaml string to the cluster
+func ApplyString(yamlString string) error {
+	cmd := kubectl("apply --server-side -f -")
+	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		return fmt.Errorf("error applying yaml: %v", err)
+	}
+
+	return nil
+}
+
+// DeleteString delete the given yaml string to the cluster
+func DeleteString(yamlString string) error {
+	cmd := kubectl("delete -f -")
+	output, err := shell.ExecuteCommandWithInput(cmd, yamlString)
+	if err != nil {
+		// Workaround because the resource can be already deleted
+		if strings.Contains(output, "not found") {
+			return nil
+		}
+
+		return fmt.Errorf("error deleting yaml: %v", err)
+
+	}
+
+	return nil
+}
+
+// GetJSONCondition returns the condition of a resource
+func GetJSONCondition(ns, resourceType, resourceName string) ([]r.Condition, error) {
+	var resource r.Resource
+
+	output, err := GetJSON(ns, resourceType, resourceName)
+	if err != nil {
+		return []r.Condition{}, err
+	}
+
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return []r.Condition{}, err
+	}
+
+	return resource.Status.Conditions, nil
+}
+
+// GetPodPhase returns the phase of a pod
+func GetPodPhase(ns, selector string) (string, error) {
+	var resource r.Resource
+
+	podName, err := GetPodName(ns, selector)
+	if err != nil {
+		return "", err
+	}
+
+	output, err := GetJSON(ns, "pod", podName)
+	if err != nil {
+		return "", err
+	}
+
+	err = json.Unmarshal([]byte(output), &resource)
+	if err != nil {
+		return "", err
+	}
+
+	return resource.Status.Phase, nil
+}
+
+// GetCRDs returns all the CRDs names in a list
+func GetCRDs() ([]string, error) {
+	cmd := kubectl("get crds -o jsonpath={.items[*].metadata.name}")
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return []string{}, fmt.Errorf("error getting crds: %v", err)
+	}
+
+	return strings.Split(output, " "), nil
+}
+
+// GetJSONList returns a json list of the resources of a namespace
+func GetJSONList(ns string) (r.ResourceList, error) {
```

Comment:
same here. I thought you commented that it doesn't return JSON, but a ResourceList, so you'd name it `GetResourceList`?